src/ResourceManagement/Compute/EncryptionSettings.cs

```diff
@@ -0,0 +1,187 @@
+                    Dictionary<string, object> protectedSettings = new Dictionary<string, object>();
+                    // Legacy-Encrypt-Extension requires AAD credentials (AADClientID in PublicSettings & AADClientSecret in ProtectedSettings) to access KeyVault.
+                    protectedSettings.Add("AADClientSecret", config.AadSecret());
+                    if (config.OsType() == OperatingSystemTypes.Linux
+                    && config.LinuxPassPhrase() != null)
+                    {
+                        protectedSettings.Add("Passphrase", config.LinuxPassPhrase());
+                    }
+                    return protectedSettings;
+                }
+                else
+                {
+                    // No protected settings for NoAAD-Encrypt-Extension.
+                    return new Dictionary<string, object>();
+                }
+            }
+
+            internal override IDictionary<string, object> ExtensionPublicSettings()
+            {
+                Dictionary<string, object> publicSettings = new Dictionary<string, object>();
+                publicSettings.Add("EncryptionOperation", "EnableEncryption");
+                publicSettings.Add("KeyEncryptionAlgorithm", config.VolumeEncryptionKeyEncryptAlgorithm());
+                publicSettings.Add("KeyVaultURL", config.KeyVaultUrl()); // KeyVault to hold "Disk Encryption Key".
+                publicSettings.Add("VolumeType", config.VolumeType().ToString());
+                publicSettings.Add("SequenceVersion", Guid.NewGuid());
+                if (config.KeyEncryptionKeyURL() != null)
+                {
+                    publicSettings.Add("KeyEncryptionKeyURL", config.KeyEncryptionKeyURL()); // KeyVault to hold Key for encrypting "Disk Encryption Key" (aka kek).
+                }
+                if (this.RequestedForLegacyEncryptExtension())
+                {
+                    // Legacy-Encrypt-Extension requires AAD credentials (AADClientID in PublicSettings & AADClientSecret in ProtectedSettings) to access KeyVault.
+                    publicSettings.Add("AADClientID", config.AadClientId());
+                }
+                else
+                {
+                    // Without AAD credentials (AADClientID in PublicSettings & AADClientSecret in ProtectedSettings) to access KeyVault,
+                    // ARM resource id of KeyVaults are required.
+                    //
+                    publicSettings.Add("KeyVaultResourceId", config.KeyVaultId());
+                    if (config.KeyEncryptionKeyURL() != null && config.KeyEncryptionKeyVaultId() != null)
+                    {
+                        publicSettings.Add("KekVaultResourceId", config.KeyEncryptionKeyVaultId());
+                    }
+                }
+                return publicSettings;
+            }
+
+            /// <return>True if user requested for Legacy-Encrypt-Extension.</return>
+            internal bool RequestedForLegacyEncryptExtension()
```

Comment:
Is this method necessary?

Reply:
It is not necessary, but I think it could be more readable when using the method